### DISCUSSION TITLE
Blog file's name

### DISCUSSION THREAD

**Discussion by minhuyenhoang** [scambifestival/dfg.scambi.org#120] [Q&A]
@xplosionmind For now, the blog's file name follow this structure: `year-month-day-Blog-Title.md`. Will the file name format stay the same for the future? Or do you have a prefer format to name the blog file?

You can name the files as what you want for their URL links. The current code can take the file names and use it as URL.

**Comment by xplosionmind** [comment]
I personally like and use clean and short URLs, also because [Derek Sivers](https://sive.rs 'Derek Sicers’ official website') (I am a huge fan) [suggests it](https://sive.rs/su 'Short URLs, Why and How — Derek Sivers').

That said, for blog posts I would suggest using **the slug of the title** as the permalink, with **no date**.

Nevertheless, I would give to anyone who creates any page (not only in the blog) the possibility of choosing **an arbitrary `permalink`**. For example:

```
---
title: The quest for finding an equilibrium
permalink: /finding-equilibrium
---
```

If it is possible to do so, it would be great.

**Reply by minhuyenhoang** [reply]
My current implementation is to use the file name as the `permalink`. For example, this file `2021-10-13-The-Paneurethic-Universe-of-Scambi.md` can be named `paneurethic.md`, and when the website create a page, it will take the file name as the `link` -> `scambi.org/blog/paneurethic`.

What do you think?

**Reply by xplosionmind** [reply]
The problem of just changing the file name is that locally files are not sorted chronologically and it becomes a mess!

Furthermore, please do not add `/blog/` in the URL, also because some posts are the only places where some notions are explained and it is useful to have them ready! (e.g. in the Markdown files I emailed you `/paneurethic`)

**Reply by minhuyenhoang** [reply]
Ah, so the way my implementation work is that it will use the file name as the URL, and the `date` specified in the `frontmatter` to sort the posts. Let me double check everything and create a PR to demonstrate this.

And yep, no `/blog/` will be in the URL.

**Comment by minhuyenhoang** [comment]
@xplosionmind Here is my implementation: PR #125 
With this, you don't need to have `permalink` in the file. Just simply name the file what you want as an URL. And the posts are guaranteed to be sorted by most recent to last recent.
Please check it out and let me know!

**Reply by xplosionmind** [reply]
I am trying to check, but build is failing (#126). 

Nevertheless, the sorting I am referring to is the sorting of markdown files, inside the website’s code, not in the front end! It would be nice to have all the `.md` files displayed chronologically also in the local blog folder.

**Reply by minhuyenhoang** [reply]
> but build is failing (https://github.com/scambifestival/scambi.org/issues/126). 

The local image folder was removed via the operation you sent, and I still have some code in there using the path, that's why it fails. Let me get it fixed real quick.

> Nevertheless, the sorting I am referring to is the sorting of markdown files, inside the website’s code, not in the front end!

I'm not quite sure I don't understand this. Inside the code, I have a function/method to sort the files before read them. Below is the code I used with comments for easier understanding.

```
export async function getStaticProps() {
	const files = getAllPosts();         // here I load all the files

        // here is where I read each file
	let posts = files.map((post) => {
		const readFile = getAPost(post);
		let data = null;
		if (readFile) {
			const parse = matter(readFile);     // this is where the metadata is read
			data = parse.data;        // this is where the actual Markdown content is read
			data.date = new Date(data.date).toLocaleDateString();       // here I transform the date into the format on Figma
		}
		post = post.replace('.md', '');   

		return {
			path: post,
			frontmatter: data,
		};
	});

	posts.sort(sortByDate);    // here is where all files get sorted by date
```
